### PR TITLE
[Feature]add fiav2 into acl_graph

### PR DESCRIPTION
**Pull request by shikang-hangzhou**
-->
There are client requires adaptation and optimization of the GPT-OSS model. The attention structure adopted by this model contains the sinks parameter. The current FIA operator does not support attention with the sinks parameter, leaving only the FIAv2 operator as an alternative. However, the FIAv2 operator has not been adapted for acl graph mode. This PR enables graph for FIAv2 to achieve performance optimization.
### How was this patch tested?

- vLLM version: v0.20.1
- vLLM main: https://github.com/vllm-project/vllm/commit/c7aa186d67b6f051680831418e957c67f34ba7a2

### CODE REVIEW

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -726,6 +805,100 @@ def full_graph_fia(
...
 
+    def full_graph_fia_v2(
+        self,
+        query: torch.Tensor,
+        key: torch.Tensor,
+        value: torch.Tensor,
+        attn_metadata: AscendMetadata,
+        output: torch.Tensor,
+    ) -> torch.Tensor:
+        key, value, block_size, block_table, actual_seq_lengths_kv = self._get_fia_params(key, value, attn_metadata)
```
Comment (nakairika) [inline vllm_ascend/attention/attention_v1.py:833]: If _get_fia_params already returns actual_seq_lengths_kv, the subsequent assignment makes the first return value meaningless.
Suggestion: Either use the return value from _get_fia_params directly or remove the redundant second assignment.

File: vllm_ascend/worker/model_runner_v1.py [hunk 2]
```diff
@@ -3159,6 +3162,13 @@ def mock_true():
             if self.eplb_enable:
                 self.vllm_config.parallel_config.enable_eplb = True
             self.model: nn.Module = get_model(vllm_config=self.vllm_config)
+            for name, _ in self.model.named_parameters():
+                # sinks is a kind of parameter in attention
+                # only set in weight name
+                # TODO: remove it when fia merge in fiav2
+                if "sink" in name:
```
Comment (nakairika) [inline vllm_ascend/worker/model_runner_v1.py:3321]: Using "sink" in name as the detection condition is too broad and may falsely match parameters containing "sink" as a substring (e.g., "sinking", "unsinkable").
Suggestion: Use more precise pattern matching such as name.endswith(".sink") or "sink_weight" in name.

File: vllm_ascend/worker/model_runner_v1.py [hunk 3]
```diff
@@ -1877,6 +1878,7 @@ def execute_model(
                 model_instance=self.model,
                 max_tokens_across_pcp=0 if self.pcp_size == 1 else self.pcp_manager.max_num_tokens_across_pcp,
                 skip_compiled=has_encoder_input,
+                has_sinks = self._has_sinks,
```
Comment (nakairika) [inline vllm_ascend/worker/model_runner_v1.py]: While the current instances are correct, please ensure consistent spacing around = operators throughout the file.

File: vllm_ascend/worker/model_runner_v1.py [hunk 4]
```diff
@@ -3159,6 +3162,13 @@ def mock_true():
             if self.eplb_enable:
                 self.vllm_config.parallel_config.enable_eplb = True
             self.model: nn.Module = get_model(vllm_config=self.vllm_config)
+            for name, _ in self.model.named_parameters():
+                # sinks is a kind of parameter in attention
+                # only set in weight name
+                # TODO: remove it when fia merge in fiav2
```
Comment (nakairika) [inline vllm_ascend/worker/model_runner_v1.py:3320]: This TODO appears in multiple locations but lacks a specific merge plan or tracking issue.
Suggestion: Create a related issue and reference it in the TODO, or specify an estimated completion timeline.

File: vllm_ascend/worker/model_runner_v1.py [hunk 5]
```diff
@@ -262,6 +262,7 @@ def __init__(self, vllm_config: VllmConfig, device: torch.device):
         # gdn_query_start_loc is an unpadded version of query_start_loc.
         # TODO delete it if fia's check is removed.
         self._has_gdn = check_gdn_layer(vllm_config)
+        self._has_sinks = False
```
Comment (nakairika) [inline vllm_ascend/worker/model_runner_v1.py:277]: _has_sinks is initialized to False in __init__, but the actual detection logic is in mock_true (after model loading).
If mock_true is not called (in certain code paths), _has_sinks remains False, which may cause functional issues.
Suggestion: Ensure sink detection is performed in all model loading paths, or move the detection logic to a more appropriate location.

File: vllm_ascend/attention/attention_v1.py [hunk 6]
```diff
@@ -862,10 +1035,14 @@ def forward_fused_infer_attention(
         # runner v2, there is not capturing attribute in forward_context,
         # just use getattr to avoid attribute error.
         if _EXTRA_CTX.capturing:
-            attn_output, num_tokens = self.full_graph_fia(query, key, value, attn_metadata, output)
-            output[:num_tokens] = attn_output[:num_tokens]
-            return output
-
+            if self.sinks is not None:
```
Comment (nakairika) [inline vllm_ascend/attention/attention_v1.py:1055]: self.sinks is typed as Optional[torch.Tensor], but in full_graph_fia_v2 it's directly passed as learnable_sink parameter.
Suggestion: Add type checking or assertions to ensure self.sinks is a valid tensor before passing to the NPU operator.

File: tests/ut/attention/test_attention_v1.py [hunk 7]
```diff
@@ -190,6 +190,20 @@ def setUp(self):
...
+            head_size=64,
+            scale=1.0,
+            num_kv_heads=8,
+            alibi_slopes=None,
+            sliding_window=1024,
+            kv_cache_dtype="float16",
+            logits_soft_cap=None,
+            attn_type=self.attention_type.DECODER,
+            kv_sharing_target_layer_name=None,
+            sinks=torch.tensor([-3.4062], dtype=torch.bfloat16),
```
Comment (nakairika) [inline tests/ut/attention/test_attention_v1.py:204]: Using negative values for sink may not reflect actual model parameter ranges.
Suggestion: Use values closer to actual model parameters, or add comments explaining the source of test values.

File: tests/ut/attention/test_attention_v1.py [hunk 8]
```diff
@@ -190,6 +190,20 @@ def setUp(self):
             kv_sharing_target_layer_name=None,
         )
 
+        self.impl_swa_sink = AscendAttentionBackendImpl(
```
Comment (nakairika) [inline tests/ut/attention/test_attention_v1.py:193]: Missing tests for the edge case where _has_sinks=True but sinks is None in the attention implementation.

File: docs/source/tutorials/models/gpt-oss-120b.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+# To reduce memory fragmentation and avoid out of memory
+export HCCL_OP_EXPANSION_MODE="AIV"
+export HCCL_BUFFSIZE=512
+export NPU_MEMORY_FRACTION=0.95
+export ASCEND_RT_VISIBLE_DEVICES=0,1,2,3
+export OMP_PROC_BIND=false
+export VLLM_USE_V1=1
+export TASK_QUEUE_ENABLE=1
+export OMP_NUM_THREADS=1
+export TIKTOKEN_ENCODINGS_BASE=/{PWD}/tiktoken_encodings
```
Comment (nakairika) [inline docs/source/tutorials/models/gpt-oss-120b.md]: /{PWD} should be ${PWD} or $PWD.
Suggestion: Fix to export TIKTOKEN_ENCODINGS_BASE=${PWD}/tiktoken_encodings.

File: docs/source/tutorials/models/gpt-oss-120b.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+--tensor-parallel-size 4 \
+--max-model-len 4096 \
+--max-num-batched-tokens 4096 \
+--enable-expert-parallel \
+--compilation_config '{"cudagraph_mode": "FULL_DECODE_ONLY", "cudagraph_capture_sizes":[1,2,3,4]}'
+```
+
+The parameters are explained as follows:
+
+- `--tensor-parallel-size` are common settings ford tensor parallelism (TP) sizes.
```
Comment (nakairika) [inline docs/source/tutorials/models/gpt-oss-120b.md]: "ford" should be "for".

File: tests/ut/attention/test_attention_v1.py (hunk 8 above)
Comment (shikang-hangzhou) [reply tests/ut/attention/test_attention_v1.py:193]: > Missing tests for the edge case where _has_sinks=True but sinks is None in the attention implementation.

this is eage case and graph case is unavailable in UT now.

File: docs/source/tutorials/models/gpt-oss-120b.md (hunk 9 above)
Comment (shikang-hangzhou) [reply docs/source/tutorials/models/gpt-oss-120b.md]: > /{PWD} should be ${PWD} or $PWD. Suggestion: Fix to export TIKTOKEN_ENCODINGS_BASE=${PWD}/tiktoken_encodings.

Thank your reply, I will fix it.

File: docs/source/tutorials/models/gpt-oss-120b.md (hunk 10 above)
Comment (shikang-hangzhou) [reply docs/source/tutorials/models/gpt-oss-120b.md]: > "ford" should be "for".

Thank your reply, I will fix it.

File: vllm_ascend/attention/attention_v1.py (hunk 1 above)
Comment (shikang-hangzhou) [reply vllm_ascend/attention/attention_v1.py:833]: > If _get_fia_params already returns actual_seq_lengths_kv, the subsequent assignment makes the first return value meaningless. Suggestion: Either use the return value from _get_fia_params directly or remove the redundant second assignment.

its a good suggestion, but I’m not sure if such modifications will affect other models, so I still keep the original implementation.

File: vllm_ascend/worker/model_runner_v1.py (hunk 2 above)
Comment (shikang-hangzhou) [reply vllm_ascend/worker/model_runner_v1.py:3321]: > Using "sink" in name as the detection condition is too broad and may falsely match parameters containing "sink" as a substring (e.g., "sinking", "unsinkable"). Suggestion: Use more precise pattern matching such as name.endswith(".sink") or "sink_weight" in name.

This implementation follows the same pattern as the vLLM codebase to ensure consistency.

File: vllm_ascend/worker/model_runner_v1.py (hunk 3 above)
Comment (shikang-hangzhou) [reply vllm_ascend/worker/model_runner_v1.py]: > While the current instances are correct, please ensure consistent spacing around = operators throughout the file.

Thank your reply, I will fix it.

File: tests/ut/attention/test_attention_v1.py (hunk 7 above)
Comment (shikang-hangzhou) [reply tests/ut/attention/test_attention_v1.py:204]: > Using negative values for sink may not reflect actual model parameter ranges.
> Suggestion: Use values closer to actual model parameters, or add comments explaining the source of test values.

its actual model parameters

**Review by weijinqian0 — APPROVED** [review]
